### PR TITLE
Only PR builds to run travis tests

### PR DESCRIPTION
**Pull request by OwenArnold**
Should save time for master builds and will mirror the current Appveyor settings.

Only PRs will run build test executables and run them.

### CODE REVIEW

**Comment by OwenArnold** [comment]
@DanNixon that might indeed be neater.

File: .travis.yml [hunk 1]
```diff
@@ -28,19 +28,19 @@ jobs:
     - stage: "Unit tests"
       compiler: gcc
       name: "gcc"
-      if: NOT type = cron
+      if: branch = master AND NOT type = pull_request
```
Comment (DanNixon) [inline .travis.yml]: Is `NOT TYPE` correct here?

Comment (OwenArnold) [reply .travis.yml]: Thanks. Should be corrected now to match the others in the "Unit Test" stage.

File: .travis.yml [hunk 2]
```diff
@@ -28,19 +28,19 @@ jobs:
     - stage: "Unit tests"
       compiler: gcc
       name: "gcc"
-      if: NOT type = cron
+      if: NOT branch = master AND type = pull_request
```
Comment (DanNixon) [inline .travis.yml]: Does the branch specifier need to be here? Should `type = pull_request` not be enough?

(for reference when `type = pull_request`, `branch` is the target/base branch (https://docs.travis-ci.com/user/conditions-v1#integration))

**Pull request merged**